File: solutions/uri/1914/1914.cpp
#include <iostream>
#include <string>

int main() {
    int qt;
    long first_number, second_number;
    bool is_even, first_won;
    std::string first_player, first_choice, second_player, second_choice;

    std::cin >> qt;

    while (qt--) {
        std::cin >> first_player >> first_choice >> second_player
            >> second_choice;
        std::cin >> first_number >> second_number;

        is_even = (first_number + second_number) % 2 == 0;
        first_won
            = ((is_even && first_choice == "PAR")
               || (!is_even && first_choice == "IMPAR"));

        std::cout << (first_won ? first_player : second_player) << std::endl;
    }
    return 0;
}

File: solutions/beecrowd/1914/1914.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

qt = int(input())

for _ in range(qt):
    first_player, first_choice, second_player, second_choice = (
        input().strip().split()
    )
    first_number, second_number = map(int, input().split())

    is_even = (first_number + second_number) % 2 == 0
    first_won = (is_even and first_choice == 'PAR') or (
        not is_even and first_choice == 'IMPAR'
    )

    output(f'{first_player if first_won else second_player}\n')
